data_generation.py

```diff
@@ -49,6 +49,6 @@
     }
 
     jsonString = json.dumps(dane)
-    jsonFile = open("data.json", "w")
-    jsonFile.write(jsonString)
-    jsonFile.close()
\ No newline at end of file
+    with open("data.json", "w", encoding="utf-8") as jsonFile:
+        jsonFile.write(jsonString)
+        jsonFile.close()
```

Comment:
When you use the "with" construction it closes automatically so you don;t need this line.